index.js

```diff
@@ -49,37 +49,40 @@ class AddressFinder extends React.Component {
     const { chosenAddress, results, error, searchTerm } = this.state;
     return (
       <div>
-        <div>Address finder:</div>
+        <label htmlFor="search">Address finder:</label>
         <div className="input_box">
           <input
+            id="search"
+            name="search"
             className="input"
             value={searchTerm}
             onChange={this.handleChange}
           />
           {searchTerm.length !== 0 && (
-            <div
+            <button
               className="clear_button"
               onClick={() => this.clearSearchValue()}
             >
-              <img src="clear.svg" />
-            </div>
+              <img src="clear.svg" alt="Clear address finder" />
+            </button>
           )}
         </div>
         {error && <div className="error">Something is wrong!</div>}
         {results.length > 0 && (
-          <div className="results">
+          <ul className="results">
             {results.map(result => (
-              <div
-                onClick={() => this.pickAddress(result)}
-                className="a_result"
-                key={result}
-              >
-                {result}
-              </div>
+              <li key={result}>
+                <button
+                  onClick={() => this.pickAddress(result)}
+                  className="a_result"
+                >
+                  {result}
+                </button>
+              </li>
             ))}
-          </div>
+          </ul>
         )}
-        {chosenAddress && <div className="chosen_address">{chosenAddress}</div>}
+        {chosenAddress && <p className="chosen_address">{chosenAddress}</p>}
```

Comment:
a `p`aragraph is simply more meaningful than a `div`. I considered using the HTML5 tag `address` here, but MDN's usages notes suggest this is probably the wrong context: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/address . It's well worth exploring new tags to find a relevant option.